### DISCUSSION TITLE
How to get user's guilds in Discord provider.

### DISCUSSION THREAD

**Discussion by ghost** [nextauthjs/next-auth#666] [Help]
Is it possible to get user's guilds from the Discord provider?

**Comment by LoriKarikari** [comment]
You can add the `guilds` scope to your provider to get the user's guilds.

```
providers: [
  Providers.Discord({
    clientId: process.env.DISCORD_CLIENT_ID,
    clientSecret: process.env.DISCORD_CLIENT_SECRET,
    scope: "guilds"
  })
}
```

**Comment by ghost** [comment]
@LoriKarikari But still the user object does not return the guild array.

**Comment by DevNvll** [comment]
> @LoriKarikari But still the user object does not return the guild array.

You have to retrieve the access token from the user discord account in your database and then use it as a bearer token to GET `https://discord.com/api/users/@me/guilds`

**Comment by ghost** [comment]
@DevNvll I did and although I have edited profile property but it still doesn't give. It only gives name and avatar, not even an user ID.
Note: When I try to get it in session callback, it always gives 401. The access token expires immediately.

**Comment by DevNvll** [comment]
> @DevNvll I did and although I have edited profile property but it still doesn't give. It only gives name and avatar, not even an user ID.
> Note: When I try to get it in session callback, it always gives 401. The access token expires immediately.

You mean you trying to get through getSession()? What I usually do is get the accessToken from the Accounts table/collection. (Assuming you have a DB configured)

This is my callbacks to return our user id:

```javascript
  callbacks: {
    session: async (session, user) => {
      session.user.id = user.id
      return Promise.resolve(session)
    }
  }
```
Now I have to use this id to retrieve the accessToken from the user discord account object in our database.
This works if you configured the session callback to return the user id like I did.
![image](https://user-images.githubusercontent.com/3536261/93672102-ab995f00-fa7e-11ea-9387-2936ccc2200a.png)

**Reply by realdocweird** [reply]
tu devi bruciare all'inferno che stai su github e metti le immagini al posto del codice. Ah si sto parlando italiano, così devi tradurre come noi stronzi dobbiamo riscrivere <3

**Comment by stale** [comment]


**Reply by nodgear** [reply]
Still relevant
